src/components/Text/Text.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react'
+import classNames from 'classnames';
+import styles from './Text.module.scss';
+
+export type TextProps = {
+    /** Дополнительный класс */
+    className?: string;
+    /** Стиль отображения */
+    view?: 'title' | 'button' | 'p-20' | 'p-18' | 'p-16' | 'p-14';
+    /** Html-тег */
+    tag?:  'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'p' | 'span';
+    /** Начертание шрифта */
+    weight?: 'normal' | 'medium' | 'bold';
+    /** Контент */
+    children: React.ReactNode;
+    /** Цвет */
+    color?: 'primary' | 'secondary' | 'accent';
+    /** Максимальное кол-во строк */
+    maxLines?: number;
+};
+
+const Text: React.FC<TextProps> = ({className='', view, tag: Tag = 'div', weight, children, color, maxLines}) => {
+    
+    const classes = [
+        styles['text'],
+        view ? styles[`text-${view}`] : '',
+        weight ? styles[`text-weight-${weight}`] : '',
+        color ? styles[`text-color-${color}`] : '',
+        className,
+      ]
+        .filter(Boolean)
+        .join(' ');
```

Comment:
[+] тут тоже, чтобы избавиться от проверок, можно использовать classnames
```suggestion
    const classes = classNames(
        styles['text'],
        view && styles[`text-${view}`],
        weight && styles[`text-weight-${weight}`],
        color && styles[`text-color-${color}`],
        className,
      )
```

а если определить пропсы по умолчанию, например, `view = 'p-14'`, то и части типа `view && ...` можно будет удалить